cartridges/int_klaviyo_core/cartridge/scripts/klaviyo/services.js

```diff
@@ -77,6 +77,73 @@ var KlaviyoEventService = ServiceRegistry.createService('KlaviyoEventService', {
 });
 
 
+var KlaviyoSubscribeProfilesService = ServiceRegistry.createService('KlaviyoSubscribeProfilesService', {
+
+    /**
+   * Create the service request
+   * - Set request method to be the HTTP POST method
+   * - Construct request URL
+   * - Append the request HTTP query string as a URL parameter
+   *
+   * @param {dw.svc.HTTPService} svc - HTTP Service instance
+   * @param {Object} args - Additional paramaters
+   * @returns {String} - A JSON string of the args
+   */
+    createRequest: function (svc, args) {
+        var key = Site.getCurrent().getCustomPreferenceValue('klaviyo_api_key');
+        if (!key || key == '') {
+            var logger = Logger.getLogger('Klaviyo', 'Klaviyo.core:  services.js  -  createRequest()');
+            logger.error(`KlaviyoSubscribeProfilesService failed because of a missing Klaviyo Private API key. Review key & configs for inconsistencies. Klaviyo API Key: ${key}`);
+            return;
+        }
+
+        svc.setRequestMethod('POST');
+        svc.addHeader('Authorization', 'Klaviyo-API-Key ' + key);
+        svc.addHeader('Content-type', 'application/json');
+        svc.addHeader('Accept', 'application/json');
+        svc.addHeader('revision', '2023-02-22');
```

Comment:
Can we use the most recent revision? I believe its `2023-06-15`

Reply:
That's a fair and good point. I was only looking at the subscribe service, but for the sake of time we can leave it as is